var/spack/repos/builtin/packages/globalarrays/package.py

```diff
@@ -25,42 +25,54 @@
 from spack import *
 
 
-class Globalarrays(CMakePackage):
-    """The Global Arrays (GA) toolkit provides a shared memory style
-    programming environment in the context of distributed array data
-    structures.
-    """
+class Globalarrays(AutotoolsPackage):
+    """Global Arrays (GA) is a Partitioned Global Address Space (PGAS)
+    programming model. It provides primitives for one-sided communication
+    (Get, Put, Accumulate) and Atomic Operations (read increment).
+    It supports blocking and non-blocking primtives, and supports location
+    consistency."""
 
     homepage = "http://hpc.pnl.gov/globalarrays/"
-    url = "https://github.com/GlobalArrays/ga"
+    url = "https://github.com/GlobalArrays/ga/releases/download/v5.7/ga-5.7.tar.gz"
 
-    version('master', git='https://github.com/GlobalArrays/ga', branch='master')
+    version('5.7',   'bb9a441a6b4fbb8b52b58c2d3f4cd07f')
+    version('5.6.5', '90da628dc72048deeda3f0cd095cb5b3')
+    version('5.6.4', '051901b316c9766b8ba54306bff7f6b3')
+    version('5.6.3', 'df3cf6cc8288d9f202b7fd0ea82f5491')
+    version('5.6.2', '901e4612203bac45059be524fc1abfb7')
+    version('5.6.1', '674c0ea9bf413840b1ff1e669de73fca')
+    version('5.6',   '49d7e997daed094eeb9565423879ba36')
 
-    variant('i8', default=False, description='Build with 8 byte integers')
+    variant('int64', default=False, description='Compile with 64 bit indices support')
+    variant('blas', default=False, description='Enable external BLAS')
```

Comment:
those variants have not been there previously and I still think they should not be added.

Reply:
There weren't there before because the previous version was a single-purpose package. This one supports more platforms other, including all those that PNNL believes to be the most used.

Reply:
Sure, but what’s the problem in using Spack’s build lapack/scalapack?

Reply:
With the depends_on() we're doing just that, correct? The variants are also being used to set the appropriate configure flags. We're not supplying a path to the lapack/scalapack, so it won't really be external. I removed the "external" reference from the descriptions.

Reply:
Ah, so you're saying that BLAS is optional, we either use the one installed by Spack or we use no BLAS, correct?

Reply:
> We're not supplying a path to the lapack/scalapack, so it won't really be external.

Sorry, I am a bit confused here. If Blas=false, you build without it altogether, right? If it’s true, do you link against Spack build Blas? (same for the other two)

Reply:
If the user were to build GA directly (not using Spack in this case), they could use `--with-blas` and get the system blas install. In the case of Spack, I believe that would end up being the Spack installed one. Or at least that is my intention.

If there user were to build GA directly with `--with-blas=<ARG>` they also have the option of specifying the path & linking to a specified external blas install as the argument. I don't have that supported through this Spack package.

Perhaps what I really need here is better description values for the variants?

Reply:
but you can get `--with-blas=<ARG>` this via Spack, right? You can ask for libs on all three, see for example `petsc` which uses blas/lapack.

Reply:
also see http://spack.readthedocs.io/en/latest/packaging_guide.html#blas-lapack-and-scalapack-libraries

Reply:
ps. 

> I believe that would end up being the Spack installed one. Or at least that is my intention.

you can check this after installation by looking at what the package links against. You should see Spack's build blas/lapack/scalapack if all three are enabled.